### ISSUE TITLE
Model run time and convergence issues related to large sample sizes

### ISSUE THREAD

**Issue by AshleeMikk** [brianstock/MixSIAR#310]
Hi Brian,

First off, thanks for all your work on this package. 

I am a new to using MixSIAR. I am estimating brown bear diet from a large, long-term data set and I have three possible questions I would like to explore: How diet varies by year, how diet varies by reproductive status, and how diet varies by individual. My data set consists of almost 800 records of 415 individual bears (some have been sampled multiple times over the years).

I first attempted a model that estimated diet by year and reproductive status and the test run took quite some time (I don't remember axactly how long now) but it did run, then I ran a model at normal settings, which took a couple days and did not converge. So I ran a model for the "long" run and had the same story. Finally I ran the "very long" settings and that model ran for 8 days, 1 million iterations, and failed to converge.

So I started experimenting with model settings and found that even with no covariates in my model, there is an exponential relationship between run time and sample size (i.e. a simple model with no covariates and a sample size of only 300 takes 4.6 hours to run. I have a very new computer with 16GB of RAM. My IT department is working on getting additional RAM for my computer and we may need to access a much faster computer.

In the meantime, do you have any thoughts or recommendations to get MixSIAR to run faster? Or would you recommend another package? From what I have heard from other MixSIAR users, these run times are very abnormal.

Thanks in advance,
Ashlee

**Comment by ericward-noaa** [comment]
Hi Ashlee -- sounds like a great dataset. 

I think the most likely culprit here isn't an issue of the sample size, but a conflict between the models you're trying to fit and data. The model with all animals and no covariates should have converged for example, if all animals had the same diet. But if there's groups of animals with different diets, the posterior will likely be multi-modal and MCMC will have a really hard time sampling (resulting in slow run times). 

So I'd maybe start by plotting the data or running the model for a subset of individuals?

**Comment by AshleeMikk** [comment]
Hi Brian,

Thanks for responding to my question. Finding a path forward on this has been difficult for me. I just started my PhD and never actually intended to estimate diet using stable isotopes, but fell down this rabbit hole by accident.

I have been looking at different model structures and sample sizes to try and resolve the issue. I randomly sub-set my data at 50, 100, 200, and 300 individuals for a model with no covariates, reproductive status as a random effect, and reproductive status as a fixed effect. I didn't want to over-parameterize the model in case that was the issue. I found that there is an exponential relationship between sample size fand run time. For example, using the model with reproductive status as a fixed effect, the model with 50 bears took 32 minutes, 100 bears took 35 minutes, 200 bears took 271 minutes, 300 bears took 488 minutes. These were all run under the "normal" settings and some models fully converged while others mostly converged (1 of 3 chains may not have converged or there are a few parameters with gelman > 1.1).

Have you run into these long run times previously? Is this just a reality for my data?

Any advice and guidance is wildly appreciated,

Ashlee


Ashlee Mikkelsen______________
PhD Student
Zedrosser Lab
University of South-Eastern Norway
Campus Bø
